### PR TITLE
chore: flatten configuration file by removing styleguide object

### PR DESCRIPTION
**Pull request by tatomyr**
## What/Why/How?

Closes https://github.com/Redocly/redocly-cli/issues/833

## Reference

Related to https://github.com/Redocly/marketing-site-portal/pull/986

## Check yourself

- [x] Code is linted
- [x] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Review by adamaltman — APPROVED** [review]

**Review by AntonKozachuk — APPROVED** [review]

File: packages/core/src/config/utils.ts [hunk 1]
```diff
@@ -39,13 +42,76 @@ export function transformApiDefinitionsToApis(
...
+
+    decorators,
+    oas2Decorators,
+    oas3_0Decorators,
+    oas3_1Decorators,
+
+    doNotResolveExamples: (rawConfigRest as FlatRawConfig).resolve?.doNotResolveExamples,
+  };
+
+  return {
```
Comment (RomanHotsiy) [inline packages/core/src/config/utils.ts:95]: should we add a check here and crash if there is both any of `rules` etc and `lint`?

**Review by RomanHotsiy — APPROVED** [review]
It looks good but I would add error message for mixed config.

Also, I'm not sure if we have a test for legacy config. If not, let's add one.

File: packages/core/src/config/utils.ts (hunk 1 above)
Comment (tatomyr) [reply packages/core/src/config/utils.ts:95]: Thanks, will add the error.

Comment (tatomyr) [reply packages/core/src/config/utils.ts:105]: Added: https://github.com/Redocly/redocly-cli/pull/847/commits/bc76589e0b64402d8e9f73cfd8d2dbbd225094f8?file-filters%5B%5D=.ts&show-viewed-files=true

@adamaltman could you verify the error message?

**Pull request merged**